### PR TITLE
The manifest_changed.sh doesn't need binary to update the manifests

### LINKED ISSUE #10909: Make etc/ci/manifest_changed.sh work on release builders
The `etc/ci/manifest_changed.sh` script does not work properly on release builders because it looks for the development version of servo.

Here is an example error (from http://build.servo.org/builders/linux-rel/builds/672/steps/shell_2/logs/stdio):
```
Binary path /home/servo/buildbot/slave/linux-rel/build/target/debug/servo does not exist
```

There are actually two fixes needed:

- Add
```
set -o errexit
set -o nounset
set -o pipefail
```
to our CI bash scripts so they fail if any command fails.
- Fix the script so that it doesn't look for a binary (I think the `--binary` option should do this).

### PR DESCRIPTION
**Pull request by askeing**
fix #10909

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10917)

### CODE REVIEW

File: etc/ci/manifest_changed.sh [hunk 1]
```diff
@@ -1,5 +1,12 @@
 #!/bin/bash
-./mach test-wpt --manifest-update SKIP_TESTS > /dev/null
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if `./mach test-wpt --manifest-update SKIP_TESTS > /dev/null`
```
Comment (askeing) [inline etc/ci/manifest_changed.sh]: I am not sure do we have to add "Try debug build..." and "Try release build..." at Line 6 and 8.
Does anyone have idea?

Comment (askeing) [inline etc/ci/manifest_changed.sh]: I'm not sure do we have to add "Try debug build..." and "Try release build..." at Line 6 and 8.
Any suggestion?

**Comment by aneeshusa** [comment]
@askeing Thanks for the PR! Trying to run the manifest-update (in debug mode), then trying again in release mode is not a good way to approach this:
  - Stylistically, this is a "guess and check" approach, where we should prefer to be explicit about which tests we are running.
  - This would need to be updated if we gained a [third optimization level](https://github.com/servo/servo/issues/9904).
  - We don't need to have any binaries to update the manifest!

I initially thought the best solution would be to change wptrunner so that passing --manifest-update with SKIP_TESTS would not look for a binary. But, as you discovered, SKIP_TESTS does not have any special meaning, so this is not a good change to make.

Instead, we should explicitly signal that we do not need a binary. This can be done by passing a `--binary=` argument to the `./mach` command. (Note that there's nothing after the equal sign.) Also, please add some comments to explain the functionality of `SKIP_TESTS` and `--binary=`.

**Comment by askeing** [comment]
It sounds very reasonable, thanks for your suggestion, I'll update the commit soon :)

File: etc/ci/manifest_changed.sh [hunk 2]
```diff
@@ -1,5 +1,12 @@
 #!/bin/bash
-./mach test-wpt --manifest-update SKIP_TESTS > /dev/null
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# We shouldn't need any binary at all to update the manifests.
+# Adding "SKIP_TESTS" to skip tests, and "--binary=" to skip binary check.
```
Comment (aneeshusa) [inline etc/ci/manifest_changed.sh]: Can you expand more about how SKIP_TESTS works in this comment? The name makes it clear that it's used to skip the tests, but we should document how it does so.

File: etc/ci/manifest_changed.sh [hunk 3]
```diff
@@ -1,5 +1,14 @@
 #!/bin/bash
-./mach test-wpt --manifest-update SKIP_TESTS > /dev/null
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# We shouldn't need any binary at all to update the manifests.
+# Adding "SKIP_TESTS" to skip tests, it doesn't really skip the tests.
+# It will run "run_wpt" with "'test_list': ['SKIP_TESTS']", and then pass it into wptrunner.
```
Comment (aneeshusa) [inline etc/ci/manifest_changed.sh]: This needs a little more explanation :)
```
and then pass it into wptrunner, which won't be able to find any tests named "SKIP_TESTS", and thus won't run any.
```

File: etc/ci/manifest_changed.sh [hunk 4]
```diff
@@ -1,5 +1,14 @@
...
+# Adding "--binary=" to skip binary check.
```
Comment (aneeshusa) [inline etc/ci/manifest_changed.sh]: Wording nit:
```
Adding "--binary=" to skip looking for a compiled servo binary.
```

**Comment by aneeshusa** [comment]
Thanks, @askeing! @bors-servo r+

**Pull request merged**